src/main/proto/wfa/measurement/internal/kingdom/protocol_data.proto

```diff
@@ -0,0 +1,81 @@
+// Copyright 2021 The Cross-Media Measurement Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+syntax = "proto3";
+
+package wfa.measurement.internal.kingdom;
+
+import "wfa/common/el_gamal_key.proto";
+import "wfa/common/noise_parameters.proto";
+
+option java_package = "org.wfanet.measurement.internal.kingdom";
+option java_multiple_files = true;
+
+// Protocol-specific configuration and parameters for the Liquid Legions V2
+// Protocol.
+message LiquidLegionsV2Data {
+  // Initial configurations
+  LiquidLegionsV2Config config = 1;
+  // ElGamal keys used for a certain report.
+  map<string, wfa.common.ElGamalPublicKey> duchy_keys = 2;
+}
+
+// Protocol-specific configuration for the Liquid Legions V2 Protocol.
+// Used internally in the report details. The fields should be set when a report
+// is created and then immutable.
+message LiquidLegionsV2Config {
```

Comment:
Where will this be used/stored?

We expect the intermediate noise values to part of the overall system config and for the output noise parameters to be set on a per computation basis.

Also, I don't think the kingdom needs to learn about intermediate noise params.

That said, it's fine to write the intermediate config to the database. But it probably means we need multiple protos to support this.

What do you think of dividing the noise not into reach and frequency noise, but output and intermediate noise?

Reply:
When the kingdom gets a CreatMeasurementRequest, a report will be created in the kingdom database report table. This protocolData(MetaData) will be set in the report proto. Then it will be the source of truth to describe what computation should be done for this report.

The protocolData/metaData will be used in two places.
1. when the duchy heralds pull from the kingdom, the returned GlobalComputations would be converted from the reports and contain these configs to instruct the duchies what computation to run and what parameters to use.
2. when the EDP call listRequisitions(), the return requisitions will also pick up the protocolData/metaData from the corresponding report (possible by joining the Requisition table and the Report table), the config will tell the EDP how to create the sketch.

The reason to let the kingdom set the intermediate noise params is to make sure all duchies use the same values for them. Otherwise, the result would be wrong.

Dividing to output and intermediate noise sounds good. we could have
1. EdpNoise  (publisher noise)
2. MpcNoise  (blinded histogram noise, noise for publisher noise)
3. OutputNoise (reach Dp noise, frequency Dp noise)

WDYT?

Reply:
Thanks for the explanation. Makes sense. I had a couple suggestions on names, but otherwise looks good.